kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/KafkaEventPublisher.java

```diff
@@ -0,0 +1,134 @@
+     *
+     * @param builder the {@link Builder} used to instantiate a {@link KafkaEventPublisher} instance
+     */
+    protected KafkaEventPublisher(Builder<K, V> builder) {
+        builder.validate();
+        this.kafkaPublisher = builder.kafkaPublisher;
+    }
+
+    /**
+     * Instantiate a Builder to be able to create a {@link KafkaEventPublisher}.
+     * <p>
+     * The {@link KafkaPublisher} is a <b>hard requirements</b> and as such should be provided.
+     *
+     * @param <K> a generic type for the key of the {@link KafkaPublisher}
+     * @param <V> a generic type for the value of the {@link KafkaPublisher}
+     * @return a Builder to be able to create a {@link KafkaEventPublisher}
+     */
+    public static <K, V> Builder<K, V> builder() {
+        return new Builder<>();
+    }
+
+    @Override
+    public Object handle(EventMessage<?> event) {
+        kafkaPublisher.send(event);
+        return null;
+    }
+
+    @Override
+    public boolean supportsReset() {
+        return DOES_NOT_SUPPORT_RESET;
+    }
+
+    /**
+     * Builder class to instantiate a {@link KafkaEventPublisher}.
+     * <p>
+     * The {@link KafkaPublisher} is a <b>hard requirements</b> and as such should be provided.
+     *
+     * @param <K> a generic type for the key of the {@link KafkaPublisher}
+     * @param <V> a generic type for the value of the {@link KafkaPublisher}
+     */
+    public static class Builder<K, V> {
+
+        private KafkaPublisher<K, V> kafkaPublisher;
+
+        /**
+         * Sets the {@link KafkaPublisher} to be used by this {@link EventMessageHandler} to publish
+         * {@link EventMessage} on.
+         *
+         * @param kafkaPublisher the {@link KafkaPublisher} to be used by this {@link EventMessageHandler} to publish
+         *                       {@link EventMessage} on
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder kafkaPublisher(KafkaPublisher<K, V> kafkaPublisher) {
+            assertNonNull(kafkaPublisher, "KafkaPublisher may not be null");
+            this.kafkaPublisher = kafkaPublisher;
+            return this;
+        }
+
+        /**
+         * Initializes a {@link KafkaEventPublisher} as specified through this Builder.
+         *
+         * @return a {@link KafkaEventPublisher} as specified through this Builder
+         */
+        public KafkaEventPublisher build() {
+            return new KafkaEventPublisher<>(this);
+        }
+
+        /**
+         * Validates whether the fields contained in this Builder are set accordingly.
+         *
+         * @throws AxonConfigurationException if one field is asserted to be incorrect according to the Builder's
+         *                                    specifications
+         */
+        @SuppressWarnings("WeakerAccess")
+        protected void validate() {
+            assertNonNull(kafkaPublisher, "The KafkaPublisher is a hard requirement and should be provided");
```

Comment:
```suggestion
            assertNonNull(kafkaPublisher, "The KafkaPublisher is mandatory and should be provided");
```

Reply:
In favor of consistency with all the other `validate()` methods, I think I'll leave it at `a hard requirement` for now. Additionally, this exact wording is used in the JavaDoc, which I feel is an extra argument to leave it like this. I am up for changing this at another point in time by the way.